client/gutenberg/extensions/simple-payments/edit.jsx

```diff
@@ -0,0 +1,474 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+import { __, _n, sprintf } from '@wordpress/i18n';
+import { Component, Fragment } from '@wordpress/element';
+import { compose, withInstanceId } from '@wordpress/compose';
+import {
+	ExternalLink,
+	PanelBody,
+	SelectControl,
+	TextareaControl,
+	TextControl,
+	ToggleControl,
+} from '@wordpress/components';
+import { InspectorControls } from '@wordpress/editor';
+import { withSelect } from '@wordpress/data';
+import apiFetch from '@wordpress/api-fetch';
+import classNames from 'classnames';
+import emailValidator from 'email-validator';
+import get from 'lodash/get';
+import trimEnd from 'lodash/trimEnd';
+
+/**
+ * Internal dependencies
+ */
+import { getCurrencyDefaults } from 'lib/format-currency';
+import {
+	SIMPLE_PAYMENTS_PRODUCT_POST_TYPE,
+	SUPPORTED_CURRENCY_LIST,
+} from 'lib/simple-payments/constants';
+import ProductPlaceholder from './product-placeholder';
+
+class SimplePaymentsEdit extends Component {
+	state = {
+		fieldEmailError: '',
+		fieldPriceError: '',
+		fieldTitleError: '',
+		isSavingProduct: false,
+	};
+
+	componentDidUpdate( prevProps ) {
+		const { simplePayment, attributes, setAttributes, isSelected, isLoadingInitial } = this.props;
+		const { currency, price, title, email, content, multiple } = attributes;
+
+		// @TODO check componentDidMount for the case where post was already loaded
+		if ( ! prevProps.simplePayment && simplePayment ) {
+			setAttributes( {
+				content: get( simplePayment, [ 'content', 'raw' ], content ),
+				currency: get( simplePayment, [ 'meta', 'spay_currency' ], currency || 'USD' ),
+				email: get( simplePayment, [ 'meta', 'spay_email' ], email ),
+				multiple: Boolean( get( simplePayment, [ 'meta', 'spay_multiple' ], Boolean( multiple ) ) ),
+				price: get( simplePayment, [ 'meta', 'spay_price' ], price || undefined ),
+				title: get( simplePayment, [ 'title', 'raw' ], title ),
+			} );
+		}
+
+		// Validate and save on block-deselect
+		if ( prevProps.isSelected && ! isSelected && ! isLoadingInitial && this.validateAttributes() ) {
+			this.saveProduct();
+		}
+	}
+
+	attributesToPost = attributes => {
+		const { title, content, currency, price, email, multiple } = attributes;
+
+		return {
+			title,
+			status: 'publish',
+			content,
+			featured_media: 0,
+			meta: {
+				spay_currency: currency,
+				spay_email: email,
+				spay_formatted_price: this.formatPrice( price, currency ),
```

Comment:
I've been looking into the ever controversial `spay_formatted_price`, since @simison expressed some really valid concerns about it being used to mislead the user about the actual price it'll end up paying.
The payment processor does not use that value, so the user will always be charged the price entered, but it is used for notifications and display if present.

This is the code on the `Simple_Payments_Product` class used by WordPress.com on the payment processor (called with `$formatted = false) and notifications:

```php
/**
 * Getter: `price`
 *
 * @param  boolean [$formatted] formats the price adding currency symbol, etc.
 * @return string product price
 */
public function get_price( $formatted = true ) {
	$product_data = $this->get();

	if ( is_wp_error( $product_data ) ) {
		return;
	}

	if ( ! $formatted ) {
		return $product_data['spay_price'];
	}

	if ( $product['spay_formatted_price'] ) {
		return $product_data['spay_formatted_price'];
	}

	$price = new Store_Price( $product_data['spay_price'], $product_data['spay_currency'] );
	return (string) html_entity_decode( strip_tags( $price->display( 2 ) ) );
}
```

and on the `Jetpack_Simple_Payments` class when parsing the shortcode:

```php
function format_price( $formatted_price, $price, $currency, $all_data ) {
	if ( $formatted_price ) {
		return $formatted_price;
	}
	return "$price $currency";
}
```

`Jetpack_Simple_Payments` does not have access to `Store_Price` since that class is not on Jetpack 😞 .

I think that **we should not send `spay_formatted_price` at all**. It won't cause any regressions, and will leave the formatting to the backend when really needed. I'll start a Jetpack issue to remove it completely so nobody is confused by this again.